test/e2e/kibishii_tests.go

```diff
@@ -143,3 +151,47 @@ func VerifyData(ctx context.Context, namespace string, levels int, filesPerLevel
 	}
 	return err
 }
+
+// RunKibishiiTests runs kibishii tests on the provider.
+func RunKibishiiTests(client *kubernetes.Clientset, providerName, veleroCLI, backupName, restoreName string) error {
+	fiveMinTimeout, _ := context.WithTimeout(context.Background(), 5*time.Minute)
+	oneHourTimeout, _ := context.WithTimeout(context.Background(), time.Minute*60)
+
+	if err := CreateNamespace(fiveMinTimeout, client, kibishiiNamespace); err != nil {
+		errors.Wrapf(err, "Failed to create namespace %s to install Kibishii workload", kibishiiNamespace)
+	}
+
+	if err := installKibishii(fiveMinTimeout, kibishiiNamespace, providerName); err != nil {
+		errors.Wrap(err, "Failed to install Kibishii workload")
+	}
+
+	if err := generateData(oneHourTimeout, kibishiiNamespace, 2, 10, 10, 1024, 1024, 0, 2); err != nil {
+		return errors.Wrap(err, "Failed to generate data")
+	}
+
+	if err := BackupNamespace(oneHourTimeout, veleroCLI, backupName, kibishiiNamespace); err != nil {
+		return errors.Wrapf(err, "Failed to backup kibishii namespace %s", kibishiiNamespace)
+	}
+
+	fmt.Printf("Simulating a disaster by removing namespace %s\n", kibishiiNamespace)
+	if err := client.CoreV1().Namespaces().Delete(oneHourTimeout, kibishiiNamespace, metav1.DeleteOptions{}); err != nil {
+		return errors.Wrap(err, "Failed to simulate a disaster")
+	}
+
+	if err := RestoreNamespace(oneHourTimeout, veleroCLI, restoreName, backupName); err != nil {
+		return errors.Wrapf(err, "Restore %s failed from backup %s", restoreName, backupName)
+	}
+
+	// TODO - check that namespace exists
```

Comment:
Did you want to address this TODO in this PR?

Reply:
This code was moved into this file from PR #3060 https://github.com/vmware-tanzu/velero/pull/3060/files#diff-c1cddc33537f5a0b043fc5b2e1c38f347f6e95ca915fce01b853c2fbc473a825R69

So not addressing that `TODO` as part of this PR. But looking at the code, IMO this `TODO` can be removed.
I am going to let @dsu-igeek comment

Reply:
I am going to keep this `TODO` for now and re-evaluate as we make more progress with E2@ tests.